specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/azureFirewall.json

```diff
@@ -0,0 +1,648 @@
+         },
+         "allOf":[  
+            {  
+               "$ref":"./network.json#/definitions/SubResource"
+            }
+         ],
+         "description":"Application rule collection resource"
+      },
+      "AzureFirewallApplicationRuleProtocol":{  
+         "properties":{  
+            "protocolType":{  
+               "description":"Protocol type",
+               "$ref":"#/definitions/AzureFirewallApplicationRuleProtocolType"
+            },
+            "port":{  
+               "type":"integer",
+               "format":"int32",
+               "maximum":64000,
+               "exclusiveMaximum":false,
+               "minimum":0,
+               "exclusiveMinimum":false,
+               "description":"Port number for the protocol, cannot be greater than 64000. This field is optional."
+            }
+         },
+         "description":"Properties of the application rule protocol."
+      },
+      "AzureFirewallApplicationRule":{  
+         "properties":{  
+            "name":{  
+               "type":"string",
+               "description":"Name of the application rule."
+            },
+            "description":{  
+               "type":"string",
+               "description":"Description of the rule."
+            },
+			"sourceAddresses":{  
+               "type":"array",
+               "description":"List of source IP addresses for this rule.",
+               "items":{  
+                  "type":"string"
+               }
+            },
+            "protocols":{  
+               "type":"array",
+               "items":{  
+                  "$ref":"#/definitions/AzureFirewallApplicationRuleProtocol"
+               },
+               "description":"Array of ApplicationRuleProtocols."
+            },
+            "targetUrls":{  
+               "type":"array",
+               "description":"List of URLs for this rule.",
+               "items":{  
+                  "type":"string"
+               }
+            }
+         },
+         "description":"Properties of an application rule."
+      },
+      "AzureFirewallNetworkRuleCollectionPropertiesFormat":{  
+         "properties":{  
+            "priority":{  
+               "type":"integer",
+               "format":"int32",
+               "maximum":65000,
+               "exclusiveMaximum":false,
+               "minimum":0,
+               "exclusiveMinimum":false,
+               "description":"Priority of the network rule collection resource."
+            },
+            "action":{  
+               "$ref":"#/definitions/AzureFirewallRCAction",
+               "description":"The action type of a rule collection"
+            },
+            "rules":{  
+               "type":"array",
+               "items":{  
+                  "$ref":"#/definitions/AzureFirewallNetworkRule"
+               },
+               "description":"Collection of rules used by a network rule collection."
+            },
+            "provisioningState":{  
```

Comment:
We only care about provisioning state on AzureFirewallProperties why do we have to mentioned provisioning state on these child resources? we don't have operations on them. Also the same is for Ipconfig as well.

Reply:
I see that other resources have it and NRP returns this value. So you can ignore this comment.